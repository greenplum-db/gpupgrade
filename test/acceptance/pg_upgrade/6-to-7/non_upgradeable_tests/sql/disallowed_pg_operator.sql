-- Copyright (c) 2017-2024 VMware, Inc. or its affiliates
-- SPDX-License-Identifier: Apache-2.0

--------------------------------------------------------------------------------
-- Create and setup non-upgradeable objects
--------------------------------------------------------------------------------
CREATE OPERATOR => (leftarg = int8, procedure = numeric_fac);

CREATE DATABASE test_disallowed_pg_operator;
1:@db_name test_disallowed_pg_operator:CREATE OPERATOR => (leftarg = int8, procedure = numeric_fac);
1q:

---------------------------------------------------------------------------------
--- Assert that pg_upgrade --check correctly detects the non-upgradeable objects
---------------------------------------------------------------------------------
!\retcode gpupgrade initialize --source-gphome="${GPHOME_SOURCE}" --target-gphome=${GPHOME_TARGET} --source-master-port=${PGPORT} --disk-free-ratio 0 --non-interactive;
! find $(ls -dt ~/gpAdminLogs/gpupgrade/pg_upgrade_*/ | head -1) -name "databases_with_disallowed_pg_operator.txt" -exec cat {} +;

---------------------------------------------------------------------------------
--- Cleanup
---------------------------------------------------------------------------------

DROP OPERATOR => (bigint, NONE);
DROP DATABASE test_disallowed_pg_operator;
