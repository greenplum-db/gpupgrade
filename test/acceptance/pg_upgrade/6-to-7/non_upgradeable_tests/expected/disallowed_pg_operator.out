-- Copyright (c) 2017-2024 VMware, Inc. or its affiliates
-- SPDX-License-Identifier: Apache-2.0

--------------------------------------------------------------------------------
-- Create and setup non-upgradeable objects
--------------------------------------------------------------------------------
CREATE OPERATOR => (leftarg = int8, procedure = numeric_fac);
CREATE OPERATOR

CREATE DATABASE test_disallowed_pg_operator;
CREATE DATABASE
1:@db_name test_disallowed_pg_operator:CREATE OPERATOR => (leftarg = int8, procedure = numeric_fac);
CREATE OPERATOR
1q: ... <quitting>

---------------------------------------------------------------------------------
--- Assert that pg_upgrade --check correctly detects the non-upgradeable objects
---------------------------------------------------------------------------------
!\retcode gpupgrade initialize --source-gphome="${GPHOME_SOURCE}" --target-gphome=${GPHOME_TARGET} --source-master-port=${PGPORT} --disk-free-ratio 0 --non-interactive;
-- start_ignore
-- end_ignore
(exited with code 1)
! cat ~/gpAdminLogs/gpupgrade/pg_upgrade/p-1/databases_with_disallowed_pg_operator.txt;
isolation2test
test_disallowed_pg_operator


---------------------------------------------------------------------------------
--- Cleanup
---------------------------------------------------------------------------------

DROP OPERATOR => (bigint, NONE);
DROP OPERATOR
DROP DATABASE test_disallowed_pg_operator;
DROP DATABASE
